src/utils/viewer/viewer.cpp

```diff
@@ -420,20 +419,21 @@ Key Commands:
 
   // single inline for logging agent state msgs, so can be easily modified
   inline void showAgentStateMsg(bool showPos, bool showOrient) {
-    std::stringstream strDat("");
+    std::string res("");
     if (showPos) {
-      strDat << "Agent position "
-             << Eigen::Map<esp::vec3f>(agentBodyNode_->translation().data())
-             << " ";
+      auto v = agentBodyNode_->translation();
+      Cr::Utility::formatInto(res, res.length(),
+                              "Agent position  : [{} {} {}] ", v.x(), v.y(),
+                              v.z());
```

Comment:
Unnecessarily overcomplicated, just use the builtin debug output ~~redirected to a stringstream~~. The redirection and `if(res.size() > 0)` is also redundant, if it's going to get printed it's non-empty already anyway.

```cpp
ESP_DEBUG() << "Agent position:" << agentBodyNode_->translation();
```

Same below. If you want a more compact output for vectors (and quats), use `Mn::Debug::packed << agentBodyNode_->translation()` instead.